ccsdspy/converters.py

```diff
@@ -0,0 +1,189 @@
+
+    To write a converter, one must create a subclass and override either the method
+    `convert_many(field_array)` or `convert_one(field_array)`. The
+    `convert_many(field_array)` method implements the conversion for an entire
+    sequence of decoded packet field values in a single call (this is sometimes
+    faster), while the `convert_one(field_array)` method implements the
+    conversion for a single packet field at a time (this is sometimes easier to
+    write).
+    """
+
+    def __init__(self):
+        raise NotImplementedError("This is a base class not meant to be instantiated directly")
+
+    def convert_many(self, field_array):
+        """Convert a sequence of decoded packet field values.
+
+        This default implementation loops over the elements and defers the
+        conversion to `self.convert_one(field_array[i])`.
+
+        Parameters
+        ----------
+        field_array : NumPy array
+            decoded packet field values, must have at least one dimension
+
+        Returns
+        -------
+        converted_field_array : NumPy array
+            converted form of the decoded packet field values
+        """
+        converted_field_array = []
+
+        for i in range(field_array.shape[0]):
+            converted = self.convert_one(field_array[i])
+            converted_field_array.append(converted)
+
+        # Let NumPy infer the type from the elements.
+        converted_field_array = np.array(converted_field_array)
+
+        return converted_field_array
+
+    def convert_one(self, field_value):
+        """Convert a singel decoded packet field value.
+
+        Parameters
+        ----------
+        field_value : object
+            a single decoded packet field value
+
+        Returns
+        -------
+        converted_field_value : object
+            converted form of the decoded packet field value
+        """
+        raise NotImplementedError("This method must be overridden by a subclass")
+
+
+class PolyConverter(Converter):
+    """Post-processing conversion which applies calibration using a series
+    of coefficients ordered from highest power to intercept.
+    """
+
+    def __init__(self, coeffs):
+        """Instantiate a PolyConverter object
+
+        Parameters
+        ----------
+        coeffs : list of float
+           Polynomial coefficients ordered from highest power to intercept.
+        """
+        self._coeffs = coeffs
+
+    def convert_many(self, field_array):
+        """Apply the polynomial conversion.
+
+        Parameters
+        ----------
+        field_array : NumPy array
+            decoded packet field values, must have at least one dimension
+
+        Returns
+        -------
+        converted : NumPy array
+            converted form of the decoded packet field values
+        """
+        converted = np.zeros(field_array.shape, dtype=np.float64)
+
+        for power, coeff in enumerate(reversed(self._coeffs)):
```

Comment:
why note just use [np.poly1d](https://numpy.org/doc/stable/reference/generated/numpy.poly1d.html)?

Reply:
I'll do that